asreview/webapp/app.py

```diff
@@ -41,50 +41,38 @@
 from asreview.webapp.authentication.oauth_handler import OAuthHandler
 
 
-def create_app(
-    env="development",
-    config_file=None,
-    secret_key=None,
-    salt=None,
-    enable_authentication=False,
-):
+def create_app(config_path=None):
+    """Create a new ASReview webapp.
+
+    For use with WSGI servers, such as gunicorn:
+
+        > gunicorn -w 4 -b "127.0.0.1:5006" "asreview.webapp.app:create_app()"
+
+    Returns
+    -------
+    Flask:
+        The ASReview webapp.
+    """
+
     app = Flask(
         __name__,
         instance_relative_config=True,
         static_folder="build/static",
         template_folder="build",
     )
 
-    app.config["ALLOWED_ORIGINS"] = ["http://localhost:3000", "http://127.0.0.1:3000"]
-
-    app.config["SECRET_KEY"] = secret_key
-    app.config["SALT"] = salt
+    app.config.from_prefixed_env()  # deprecated, remove in version 2.0
+    app.config.from_prefixed_env("ASREVIEW_LAB")
 
-    app.config.from_prefixed_env()
+    # load config from file
+    if config_fp := (config_path or app.config.get("CONFIG_PATH", None)):
+        app.config.from_file(Path(config_fp).absolute(), load=tomllib.load, text=False)
 
-    # If allowed origins comes as an environment variables, and is a list,
-    # then we assume a JSON string.
-    if isinstance(app.config.get("ALLOWED_ORIGINS", False), str):
-        app.config["ALLOWED_ORIGINS"] = json.loads(app.config["ALLOWED_ORIGINS"])
+    # if there are no cors and config is in debug mode, add default cors
+    if app.debug and not app.config.get("CORS_ORIGINS", None):
+        app.config["CORS_ORIGINS"] = ["http://localhost:3000", "http://127.0.0.1:3000"]
 
-    if config_file_path := config_file or app.config.get("CONFIGFILE", ""):
```

Comment:
Renamed this env var from "FLASK_CONFIGFILE" into "ASREVIEW_LAB_CONFIG_PATH". Needs backward compatility?

Reply:
Resolved